Implement Prototype

hayfield opened the pull request:
This adds a prototype that loads and parses data files, then outputs a multicoloured sortable and searchable table containing data.

hayfield committed:
Add some basic test data

This adds 3 entries with made up numbers to a csv
file that contains static data about signatories.

hayfield committed:
Add install instructions to README

hayfield committed:
Rename a file

This allows separation of static and dynamic content

hayfield committed:
Add script to load remote data

This loads the summary statistics csv and writes it to a file.

hayfield committed:
Dashes to underscores in filenames

hayfield committed:
Load data files

Two functions are provided.
One loads straight into a string.
The other parses a CSV file into an array of dictionaries.

hayfield committed:
Convert data to be publisher-queryable

hayfield committed:
Load summary_stats for Grand Bargain signatories

hayfield committed:
Generate random humanitarian values

It is OK for values to be below the baselines. This is because
values may decrease rather than increase when publishers change
data.

hayfield committed:
Add basic HTML template with styles

The content is to be expanded, though this sets something up
that works in a static manner.

hayfield committed:
Add a flask app

This renders a table with some data in.

hayfield committed:
Handle non-existant data

Something should be displayed when data is not available

hayfield committed:
Provide more sensible default values

hayfield committed:
Add a couple of TBC columns

This adds columns that could be added

hayfield committed:
Provide a customisable heading

Because 'Grand Bargain Monitoring' is a more useful heading than 'Heading'

hayfield committed:
Create a (really bad) gradient of colour

This provides a visual representation of the colours represnted.
The actual colours need changing.

hayfield committed:
Change gradient colours

They are still not nice colours, but at least the numbers can be read

hayfield committed:
Remove a print statement

So that it is not output when Flask is started up

hayfield committed:
Add bootstrap

This gives some styling to the page

hayfield committed:
Rename files and improve sorting

This improves table sorting and moves static files into a static files folder.

hayfield committed:
Make headings not-blue

hayfield committed:
Add padding around the page

hayfield committed:
Add a non-placeholder page title

hayfield committed:
Remove unused and commented code

hayfield committed:
Add comments to stylesheet

dalepotter commented on .gitignore:
```diff
@@ -1,3 +1,6 @@
+# untracked data
```
Small point, but could make the comment more descriptive - e.g. `# CSV and data files pulled from a remote source`

dalepotter commented on src/app.py:
```diff
@@ -0,0 +1,24 @@
+app = Flask(__name__)
+
+
+@app.route("/hello")
```
This route could be deleted!

dalepotter commented on src/parse_data.py:
```diff
@@ -0,0 +1,85 @@
+	Params:
+		base_info (array of dict): An array of dictionaries containing the base info.
+		summary_stats (array of dict): An array of dictionaries containing the summary stats.
+
```
I would suggest being more precise about what type of array is expected in the input params - i.e. a list or set? I don't think the function would work if an associative array was used.

hayfield replied:
Is `list` specific enough?

dalepotter commented on src/parse_data.py:
```diff
@@ -0,0 +1,85 @@
+		summary_stats (array of dict): An array of dictionaries containing the summary stats.
+
+	Returns:
+		A dictionary of dictionaries. The keys in the first-level dictionary are publisher registry IDs. Keys at the second level are names of statistics parsed from data file headers.
```
Suggest split over multiple lines to reduce line length

dalepotter commented on src/parse_data.py:
```diff
@@ -0,0 +1,85 @@
+	Loads and formats all data to be queried by publisher.
+
+	Returns:
+		A dictionary of dictionaries. The keys in the first-level dictionary are publisher registry IDs. Keys at the second level are names of statistics parsed from data file headers.
```
Same comment as above re line length :-)

hayfield committed:
Improve .gitignore clarity

hayfield committed:
Remove a test route

hayfield committed:
array -> list

Because Python uses lists

dalepotter reviewed (changes requested):
Could we fix up the docstrings as per the comments, particularity to clarify the comment re arrays before we merge?

hayfield committed:
Improve docstrings

This better uses the functionality that Google-spec docstrings provide.
In particular, in relation to return types.

hayfield committed:
Split long docstring lines

hayfield commented:
@dalepotter Changes made in response to comments.

dalepotter commented:
👍  

Looks good to me, thanks for the changes :-)
